### ISSUE TITLE
Record transfer fails if fieldnames in source have special characters

### ISSUE THREAD

**Issue by rulotec** [Myddleware/myddleware#575]
In FB I have the following field:

![image](https://user-images.githubusercontent.com/36896420/138539275-1662b855-3e69-4f51-b6a8-326080ba4fee.png)

When running a rule that uses that field in Myddleware `v2.5.7j`, it fails with this kind of error:

![image](https://user-images.githubusercontent.com/36896420/138539304-a9318401-e1f0-4d19-a0fd-9058f2041af3.png)

As you can see, there is an undefined index in an array. The index which is being tried to be used is `Â¿por_donde_le_gustarÃ­a_ser_contactado?;id;created_time`. 
```
Error : Notice: Undefined index: Â¿por_donde_le_gustarÃ­a_ser_contactado?;id;created_time /var/www/myddleware/src/Myddleware/RegleBundle/Classes/document.php Line : ( 1325 )
Error : Failed to transform data. /var/www/myddleware/src/Myddleware/RegleBundle/Classes/document.php Line : ( 1174 )
Failed to transform document : Failed to transformed data. This document is queued. /var/www/myddleware/src/Myddleware/RegleBundle/Classes/document.php Line : ( 786 )
Status : Error_transformed
```

However, in the simulation step during the rule creation, it was successful, showing the data to be transferred correctly:

![image](https://user-images.githubusercontent.com/36896420/138542429-2b0fbd96-30d4-4d0e-96e8-380c94cf1d35.png)

When doing this query: `SELECT * FROM DocumentData`, I see that the name of the field is stored like this:

```
MariaDB [mydwa_produccion]> select * from DocumentData\G
*************************** 1. row ***************************
    id: 1
doc_id: 61735660e30546.34998073
  type: S
  data: s:216:"{"email":"xxxxx@hotmail.com","full_name":"xxxxxxxxxxx","phone_number":"xxxxxx","\u00bfpor_donde_le_gustar\u00eda_ser_contactado?":"llamada","id":"403274678071597","created_time":"2021-10-22 23:50:53"}";
```
I'm guessing that during the simulation, no record is created in the DocumentData table. But during the actual transfer task, the data is first saved in the db as a first step and then retrieved to be transformed and send to the target system. Maybe when retrieving the data, the fieldname gets messed up and that's when the error happens.

**Comment by rulotec** [comment]
@Myddleware is there a chance this might get fixed soon?

**Comment by rulotec** [comment]
It seems like this was not happening back in `v2.5.4b`, where I have fields that have acute accents and have worked fine for a long time. The fields don't use the `¿` character though.

**Comment by rulotec** [comment]
Upon further inspect, I see that the json parsing of the record data is failing. It takes this ` Â¿por_donde_le_gustarÃ­a_ser_contactado?;id;created_time` as a single field when in fact it should be 3.

**Comment by Myddleware** [comment]
Hi @rulotec ,
@EstelleMyddleware will analyse your case on Monday. The most difficult point will be to reproduce your issue.
Is that possible to send us the credentieal by direct message ? You can use the contact form of our website www.myddleware.com and I will send you our direct email address.
Have a nice day,
Stephane

**Comment by rulotec** [comment]
Thanks Stéphane! I sent you an email with access credentials to your email address.

**Comment by rulotec** [comment]
After the valuable input provided by @EstelleMyddleware and some testing, the error was caused, not by the special characters, but by having three fields without a formula:

![image](https://user-images.githubusercontent.com/36896420/138733730-0e7813e8-9a4f-4f0e-adab-a61e78e28cf6.png)

However, a formula cannot be created because of #573 

Furthermore I feel this issue is something that needs to be addressed in one of two ways:

1. Prevent users from saving field mappings without a formula when more than 1 field is present.
2. Provide a "default/fallback formula" which would just concatenate fields with an empty space between them in cases like this where the user leaves the formula empty.

I personally would prefer the second approach.

**Comment by AlexMyddleware** [comment]
this issue is closed due to the upcoming release of 3.4.0 and the closing of the formula issue in https://github.com/Myddleware/myddleware/pull/1147
however, should the need arise in the upcoming release, feel free to reopen the issue and tag me if the bug still occurs
